packages/components/text-link/src/TextLink.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { HTMLProps } from 'react';
+import { cx } from 'emotion';
+import { Flex, CommonProps } from '@contentful/f36-core';
+import { styles } from './TextLink.styles';
+import { TextLinkVariant } from './types';
+import { Icon, IconComponent } from '@contentful/f36-icon';
```

Comment:
I guess `f36-icon` should be a dependency of this package, but I would argue that we need to change implementation here and require to pass the Icon component as an argument, but an icon name. 

The fact that we need to download all SVG to render a link is disturbing and we need to change that.

Reply:
Sorry I'm late here, but I'm trying to understand why these changes were implemented. 

I can't see the code as it looked during this comment so maybe I'm overlooking something here, but the type `IconComponent` is just a wrapper for React's `ExoticComponent<any> | ComponentType<any>` – why would that or the `Icon` component itself require you to download all SVGs?

Reply:
Maybe in the case if tree shaking is not working right, everything from `@contentful/f36-icon` ended up in the project bundle. Even if we import only one type and Icon component. 
But It just my guess, I didn't challenge enough that suggestion. 🤷‍♂️ 
Probably @suevalov have a better understanding.